From 1f89c527c40da4d9c7984e1ae4990cdf18198276 Mon Sep 17 00:00:00 2001
From: Greg Chadwick <gac@lowrisc.org>
Date: Thu, 27 Jun 2024 14:27:19 +0200
Subject: [PATCH] Remove debug memory access check

This check does not make sense for our cosimulation environment (the
spike debug module is not in use) and leads to false failures when
randomly generated programs want to access the spike debug memory
address range.
---
 riscv/mmu.cc | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/riscv/mmu.cc b/riscv/mmu.cc
index 7bd353166a..0c543dedda 100644
--- a/riscv/mmu.cc
+++ b/riscv/mmu.cc
@@ -116,10 +116,6 @@ reg_t reg_from_bytes(size_t len, const uint8_t* bytes)
 
 bool mmu_t::mmio_ok(reg_t addr, access_type type)
 {
-  // Disallow access to debug region when not in debug mode
-  if (addr >= DEBUG_START && addr <= DEBUG_END && proc && !proc->state.debug_mode)
-    return false;
-
   return true;
 }
 
